llama_index/program/predefined/evaporate.py

```diff
@@ -0,0 +1,111 @@
+import logging
+from typing import Any, Dict, List, Optional, Type, cast
+
+from llama_index.experimental.evaporate.base import EvaporateExtractor
+from llama_index.program.base_program import BasePydanticProgram
+from llama_index.program.llm_prompt_program import BaseLLMFunctionProgram
+from llama_index.program.openai_program import OpenAIPydanticProgram
+from llama_index.program.predefined.df import (
+    DEFAULT_ROWS_DF_PARSER_TMPL,
+    DataFrameRowsOnly,
+)
+from llama_index.prompts.prompts import Prompt
+from llama_index.schema import BaseNode
+
+logger = logging.getLogger(__name__)
+
+
+class EvaporateProgram(BasePydanticProgram[DataFrameRowsOnly]):
+    """Evaporate program. You should provide the fields you want to extract.
+    Then when you call the program you should pass in a list of training_data nodes
+    and a list of infer_data nodes. The program will call the EvaporateExtractor
+    to synthesize a python function from the training data and then apply the function
+    to the infer_data.
+    """
+
+    def __init__(
+        self,
+        pydantic_program_cls: Type[BaseLLMFunctionProgram],
+        fields_to_extract: List[str],
+        df_parser_template_str: str = DEFAULT_ROWS_DF_PARSER_TMPL,
+        training_key: str = "training_data",
+        infer_key: str = "infer_data",
+        **program_kwargs: Any,
+    ) -> None:
+        """Init params."""
+        # partial format df parser template string with column schema
+        # NOTE: hack where we use prompt class to partial format
+        orig_prompt = Prompt(df_parser_template_str)
+        column_schema = ",".join(fields_to_extract)
+        new_prompt = Prompt.from_prompt(
+            orig_prompt.partial_format(
+                column_schema=column_schema,
+            )
+        )
+
+        pydantic_program = pydantic_program_cls.from_defaults(
+            DataFrameRowsOnly, new_prompt.original_template, **program_kwargs
+        )
+        self._validate_program(pydantic_program)
+        self._pydantic_program = pydantic_program
+        self._extractor = EvaporateExtractor()
```

Comment:
i should've clarified requirements here..i'd want to reimplement evaporate logic within this module and make it self-contained (it should not import `from llama_index.experimental`) 

the experimental module should be deprecated/removed